### DISCUSSION TITLE
Nullable properties support

### DISCUSSION THREAD

**Discussion by tapika** [wjakob/nanobind#329] [General]
I would like to support nullable properties, so when assigning None from python code, property would pass nullptr as a class instance instead.
Usage could be following for example:

```py
b = ...
a.Link = b
...
a.Link = None
```

What I have tried:

```cpp
struct A {}
struct B { A* link = nullptr; }

nb::class_<A>(m, "A");
nb::class_<B>(m, "B")
    .def_prop_rw("Link",
        [](B& b) { return b.link; },
        [](B& b, A* link) { b.link = link; },
        nb::arg("link").none());
```

Based on [the documentation](https://nanobind.readthedocs.io/en/latest/api_core.html#_CPPv4I00DpEN8nanobind6class_11def_prop_rwER6class_PKcRR6GetterRR6SetterDpRK5Extra), it seems that this should be supported.

However, there's a compilation error that says:

> nb_func.h(49,39): error C2338: The number of nb::arg annotations must match the argument count!

It seems that the root cause is that the same `extra` annotations are applied to both getter and setter, while they have different number of arguments.

Is this known limitation, can feature like this be supported in future ?

**Comment by yosh-matsuda** [comment]
I had the same problem but I use a custom rw property suggested in https://github.com/wjakob/nanobind/pull/236

**Comment by wjakob** [comment]
For posterity, this feature was added in commit https://github.com/wjakob/nanobind/commit/11a9b3c318f2f262509ccf4090a57699ea860cad

**Reply by tapika** [reply]
Thanks for quick response. By quick test you changes seems to be work.
